Src/World/World.cpp

```diff
@@ -102,103 +126,130 @@ void World::applyConfig(const Config* config) {
+    io->setMousePosition(screenCenter);
+    //camera->update();
+    pickMng->update();
+}
 
-void World::draw(float interpolation, RenderType r) {
-    if (r != RenderType::UIOnly) {
-        drawPlaying(interpolation);
-        scripting->drawGame(interpolation);
-        //lol->render(PGE::Matrix4x4f::constructWorldMat(PGE::Vector3f(0, 0, 0), PGE::Vector3f(0.1, 0.1, 0.1), PGE::Vector3f(0, 0, 0)));
-    }
-
-    if (r != RenderType::NoUI) {
+// Lifecycle - Frames Per Second
+void World::startFrame(float interpolation) {
+    if (loading) {
         graphics->setDepthTest(false);
-        scripting->drawMenu(interpolation);
+        scripting->updateLoadFrame(interpolation);
         graphics->setDepthTest(true);
+        graphics->swap();
+        return;
     }
 
-    graphics->swap();
-}
-
-void World::updatePlaying(float timeStep) {
-    PGE::Vector2f center = PGE::Vector2f((float)config->getWidth(), (float)config->getHeight()) * 0.5f;
-    
-    PGE::Vector2f addAngle = (io->getMousePosition() - center) * (config->sensitivity->value / 30000.f);
-    camera->addAngle(addAngle.x, addAngle.y);
+    graphics->update();
+    graphics->resetRenderTarget();
+    graphics->clear(PGE::Color(1.f, 0.f, 1.f));
 
-    // Reset mouse to center.
-    io->setMousePosition(center);
+     // 1.f works alrighty.
 
-    // View/Projection matrix.
     camera->update();
+    pickMng->render();
 
-    pickMng->update();
-}
-
-void World::drawPlaying(float interpolation) {
-    camera->updateDrawTransform(interpolation);
+    scripting->updateEveryFrame(interpolation);
+    if (!paused) {
+        scripting->updateFrame(interpolation);
+        camera->updateDrawTransform(1.f);
+    } else {
+        camera->updateDrawTransform(1.f);
+    }
     gfxRes->setCameraUniforms(camera);
-    pickMng->render();
-}
 
-void World::quit() {
-    shutdownRequested = true;
-}
+    graphics->setDepthTest(false);
+    scripting->updateMenuFrame(interpolation);
+    graphics->setDepthTest(true);
 
-Font* World::getFont() const {
-    return largeFont;
+    //lol->render(PGE::Matrix4x4f::constructWorldMat(PGE::Vector3f(0, 0, 0), PGE::Vector3f(0.1, 0.1, 0.1), PGE::Vector3f(0, 0, 0)));
+
+    graphics->swap();
 }
+
+// World Fonts
+Font* World::getFont() const {return largeFont;}
+Font* World::getFontAppleLarge() const { return fontAppleLarge; }
+Font* World::getFontAppleSmall() const { return fontAppleLarge; }
+Font* World::getFontInconsolataLarge() const { return fontInconsolataLarge; }
+Font* World::getFontInconsolataSmall() const { return fontInconsolataSmall; }
+Font* World::getFontCrystalLarge() const { return fontCrystalLarge; }
+Font* World::getFontCrystalSmall() const { return fontCrystalLarge; }
+Font* World::getFontSubotypeLarge() const { return fontSubotypeLarge; }
+Font* World::getFontSubotypeSmall() const { return fontSubotypeLarge; }
+Font* World::getFontBoldLarge() const { return fontBoldLarge; }
+Font* World::getFontBoldSmall() const { return fontBoldLarge; }
```

Comment:
Don't forget the trailing newline.

Reply:
Added.

Also noticed the small versions return the large versions.
Woops!
I have not ended up using these fonts for much because they're very unusual.
Not the best for GUI purposes.
Still good for show, however.
Perhaps i should put some instances in the test gui area of the pause menu?

Reply:
Font loading should be generalized to allow for mods to have their own fonts.